libraries/browser-tracker-core/src/tracker/index.ts

```diff
@@ -564,42 +571,54 @@ export function Tracker(
       return Math.round(offset);
     }
 
-    /*
-     * Sets or renews the session cookie
+    /**
+     * Sets or renews the session cookie.
+     * Responsible for calling the `onSessionUpdateCallback` callback.
      */
-    function setSessionCookie() {
+    function setSessionCookie(options: SessionCookieOptions = {}) {
+      const { onSessionUpdateCallback, configAnonymousTracking, clientSession } = options;
```

Comment:
Nit 2: as we called, it is a bit confusing for me to have the `onSessionUpdateCallback` being passed here because this function is called even if the session does not change. The callback only needs to be called if `startNewIdCookieSession` is called. My preference would be to handle this outside this function. In effect it is the same thing, just it would be clearer to me personally.